frb_codegen/src/ir/ty.rs

```diff
@@ -94,6 +94,14 @@ impl IrType {
         matches!(self, RustOpaque(_))
     }
 
+    #[inline]
+    pub fn is_sync_rust_opaque(&self) -> bool {
+        match self {
+            SyncReturn(sync) => sync.clone().into_inner().is_rust_opaque(),
```

Comment:
would be great not to clone, but anyway this is just uber nit

Reply:
This is due to the specifics of the new implementation of SyncReturn. 
Since it delegates functions to the internal type. RustOpaque not in `distinct output_types`  
and have to check `SyncReturn<RustOpaque<T>>`